sdk/common/src/main/java/io/opentelemetry/sdk/resources/detectors/ServiceInstanceDetector.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.resources.detectors;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.resources.Entity;
+import io.opentelemetry.sdk.resources.EntityDetector;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+public final class ServiceInstanceDetector implements EntityDetector {
+  private static final String SCHEMA_URL = "https://opentelemetry.io/schemas/1.28.0";
+  private static final String ENTITY_TYPE = "service.instance";
+  private static final AttributeKey<String> SERVICE_INSTANCE_ID =
+      AttributeKey.stringKey("service.instance.id");
+
+  private static final UUID FALLBACK_INSTANCE_ID = UUID.randomUUID();
```

Comment:
I do not care about it being a UUID or something different. The fallback part is what worries me. To me, the code looks as if the SDK would issue instance IDs eventually. This is an issue because a) surrounding agents could also issue instance IDs b) the infrastructure could be set up to issue instance IDs later in some OTel collectors enriching the data.